### PR TITLE
SNAP-42: debugging tools

### PR DESCRIPTION
**Pull request by rupl**
## https://humanitarian.atlassian.net/browse/SNAP-42

Debugging tools that were developed to facilitate the completion of the HDX integration:

* `delay` lets you specify up to 10000 milliseconds of additional delay. For times when `page.onload` still isn't enough (lazy-loaded images and whatnot)
* `debug` will dump errors, warnings, and info from the JS console into ELK, so you can see what a problematic webpage is doing when Snap loads it.
* `block` will accept a comma-separated list of domains, and Snap will block any domain which contains any of the strings. It's greedy on purpose. When a domain is blocked, the `debug` property of the ELK logs will report as such.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ One of the following two inputs is **required**:
 - `url` — (querystring parameter) the remote URL you want to render.
 - `html` — (urlencoded form data) the URL-encoded HTML you want to render.
 
-If you do not specify either of these, Snap Service will return **`HTTP 422 Unprocessable Entity`**.
+If you do not specify either of these, or specify both, Snap Service will return **`HTTP 422 Unprocessable Entity`**.
```
Comment (cafuego) [inline README.md]: 🎉

File: app/app.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /**
- * print-api
```
Comment (cafuego) [inline app/app.js]: 🙈

File: app/app.js [hunk 3]
```diff
@@ -135,6 +138,9 @@ app.post('/snap', [
   query('logo', `Must be one of the following values: ${Object.keys(logos).join(', ')}. If you would like to use your site's logo with Snap Service, please read how to add it at https://github.com/UN-OCHA/tools-snap-service#custom-logos`).optional().isIn(Object.keys(logos)),
   query('service', 'Must be an alphanumeric string identifier (hyphens, underscores are also allowed).').optional().matches(/^[A-Za-z0-9_\-]+$/),
   query('ua', '').optional(),
+  query('delay', 'Must be an integer between 0-10000 inclusive.').optional().isInt({ min: 0, max: 10000 }),
+  query('debug', 'Must be a Boolean').optional().isBoolean(),
+  query('block', 'Must be a comma-separated list of domains (alphanumeric, hyphens, dots, commas)').optional().matches(/^[A-Za-z0-9\.\,\-]+$/),
```
Comment (cafuego) [inline app/app.js]: The comma does not need to be escaped, domain names are not ascii.

File: README.md (hunk 1 above)
Comment (emmajane) [reply README.md]: Four shalt thou not count, nor either count thou two, excepting that thou then proceed to three. Five is right out. Once the number three, being the third number, be reached, then, lobbest thou thy Holy Hand Grenade of Antioch towards thy foe, who, being naughty in My sight, shall snuff it.

File: app/app.js (hunk 3 above)
Comment (rupl) [reply app/app.js]: I just like escaping everything in my regexes because it hurts to think

**Pull request merged**